### PR TITLE
refactor(drag-drop): expose more private apis as protected

### LINKED ISSUE #14113: [DragDropModule] Expose members in CdkDrag and CdkDropList
#### What is the expected behavior?
Since drag and drop are in the CDK it makes sense to expose some of the members so they can be extended.

#### What is the current behavior?
A lot of internal logic that might be useful in derived classes is marked private.

#### What is the use-case or motivation for changing an existing behavior?

A lot of logic is sitting in `CdkDropList`, most of the methods that calculate stuff should be set to **protected**:

- **_positionCache** (property) - Will allow access to current environment the drag is in, especially the sibling items. For example, in a table when dragging the header cell we also want to move all the data cells that have the same column as the header cell. Because dragging is temporary (until commit/drop) we need to do it WITHOUT CD via direct DOM updates. Having the cache will help.

academicfragrantavocet-size_restricted

- **_reset()** - will allow resetting custom modifications in derived classes
- **_getItemIndexFromPointerPosition(...)** - Will help when wanting to prevent a drag to a position based on business rules, e.g.: In a table don't move column at index 0

honoredageddegu-size_restricted

- and others: `_isPointerNearDropContainer()`, `_getItemOffsetPx()`, `_getSiblingOffsetPx()`, `_getConnectedLists()`

The other mentioned will also help on various business constraints... I don't see any harm exposing
them as protected.

Same thing with `CdkDrag`...

#### Which versions of Angular, Material, OS, TypeScript, browsers are affected?
material 7.0.3

### PR DESCRIPTION
**Pull request by crisbeto**
Exposes more of the private drag&drop APIs as protected ones, in order to make it easier for people to implement their custom directives based on the CDK ones. Also moves out some of the private helper methods into functions.

Fixes #14113.

### CODE REVIEW

File: src/cdk/drag-drop/drag.ts [hunk 1]
```diff
@@ -321,13 +321,13 @@ export class CdkDrag<T = any> implements AfterViewInit, OnDestroy {
   }
 
   /** Gets only handles that are not inside descendant `CdkDrag` instances. */
-  private getChildHandles() {
+  protected _getChildHandles() {
```
Comment (jelbourn) [inline src/cdk/drag-drop/drag.ts]: We might be inconsistent about this, but the coding standards doc says no prefix for `protected`

Comment (crisbeto) [reply src/cdk/drag-drop/drag.ts]: Doing a quick code search shows that most of the protected APIs we have at the moment are underscored.

Comment (jelbourn) [reply src/cdk/drag-drop/drag.ts]: Ack, let's revisit this another time

**Comment by jelbourn** [comment]
@crisbeto API golds need to be updated

**Comment by mmalerba** [comment]
@crisbeto needs rebase

**Comment by crisbeto** [comment]
Rebased @mmalerba.

**Comment by crisbeto** [comment]
Closing this PR since it'll increase the API surface a lot and it won't be necessary anymore once we add a service around https://github.com/angular/material2/pull/14350.

**Pull request closed without merging**